any2fasta/0.4.2/Dockerfile

```diff
@@ -1,34 +1,90 @@
-
-# install dependencies via apt-get or yum if using a centos or fedora base
-RUN apt-get update && apt-get install -y wget perl && \
-    apt-get autoclean && rm -rf /var/lib/apt/lists/*
-
-# install any2fasta
-RUN wget https://github.com/tseemann/any2fasta/archive/refs/tags/v${ANY2FASTA_VERSION}.tar.gz && \
-    tar -xvf v${ANY2FASTA_VERSION}.tar.gz && \
-    rm v${ANY2FASTA_VERSION}.tar.gz && \
-    cd any2fasta-${ANY2FASTA_VERSION} && \
-    chmod +x any2fasta
-
-# set environmental variables e.g. $PATH and locale settings for singularity compatibility
-ENV PATH="/any2fasta-${ANY2FASTA_VERSION}:$PATH" \
-    LC_ALL=C
-
-# set working directory
-WORKDIR /data
+# base image
+FROM ubuntu:focal as app
+
+# metadata
+LABEL base.image="ubuntu:focal"
+LABEL dockerfile.version="1"
+LABEL software="any2fasta"
+LABEL software.version="0.4.2"
+LABEL description="Turns files into fastas."
+LABEL website="https://github.com/tseemann/any2fasta"
+LABEL license="https://github.com/tseemann/any2fasta/blob/master/LICENSE"
+LABEL maintainer="Erin Young"
+LABEL maintainer.email="eriny@utah.gov"
+
+# version to build with
+ARG ANY2FASTA_VERSION=0.4.2
+
+# install dependencies via apt-get or yum if using a centos or fedora base
+RUN apt-get update && apt-get install -y wget perl && \
+    apt-get autoclean && rm -rf /var/lib/apt/lists/*
+
+# install any2fasta
+RUN wget https://github.com/tseemann/any2fasta/archive/refs/tags/v${ANY2FASTA_VERSION}.tar.gz && \
+    tar -xvf v${ANY2FASTA_VERSION}.tar.gz && \
+    rm v${ANY2FASTA_VERSION}.tar.gz && \
+    cd any2fasta-${ANY2FASTA_VERSION} && \
+    chmod +x any2fasta
+
+# set environmental variables e.g. $PATH and locale settings for singularity compatibility
+ENV PATH="/any2fasta-${ANY2FASTA_VERSION}:$PATH" \
+    LC_ALL=C
+
+# set working directory
+WORKDIR /data
+
+#######################
+##### NEXT STAGE! #####
+#######################
+
+# running a 'test' in the 'test' stage
+FROM app as test
+
+# install zip unzip via apt-get or yum if using a centos or fedora base
+RUN apt-get update && apt-get install -y zip unzip && \
+    apt-get autoclean && rm -rf /var/lib/apt/lists/*
+
+# check version and print help options
+RUN any2fasta -v && \
+    any2fasta -h && \
+    any2fasta -h | grep github
+
+WORKDIR /test
+
+RUN echo "downloading test data..." && \
+    wget https://github.com/tseemann/any2fasta/raw/master/test.clw && \
+    wget https://github.com/tseemann/any2fasta/raw/master/test.embl && \
+    wget https://github.com/tseemann/any2fasta/raw/master/test.fna && \
+    wget https://github.com/tseemann/any2fasta/raw/master/test.fq && \
+    wget https://github.com/tseemann/any2fasta/raw/master/test.gbk && \
+    wget https://github.com/tseemann/any2fasta/raw/master/test.gfa && \
+    wget https://github.com/tseemann/any2fasta/raw/master/test.gff && \
+    wget https://github.com/tseemann/any2fasta/raw/master/test.noseq.gff && \
+    wget https://github.com/tseemann/any2fasta/raw/master/test.sth

```

Comment:
These test files are actually already in the image at /any2fasta-${ANY2FASTA_VERSION}. I don't think you should download them again.

Reply:
done